libafl_bolts/src/tuples.rs

```diff
@@ -731,6 +731,43 @@ where
     }
 }
 
+/// Trait for structs which are capable of mapping a given type to another.
+pub trait Mapper<T> {
+    /// The result of the mapping operation.
+    type Output;
+
+    /// The actual mapping operation.
+    fn map(&mut self, from: T) -> Self::Output;
+}
+
+/// Map all entries in a tuple to another type, dependent on the tail type.
+pub trait Map<M> {
```

Comment:
Why have a different trait that has exactly the same method and signature as the one above? Seems redundant and could lead to headaches

Reply:
It is not the same signature; the one above is the *mapping operation* (that is, the functor); this one is the trait which says that a particular *tuple* may be mapped.

Reply:
Well the signature is identical, just the names are slightly different :D

Reply:
`fn map_tuple` or get rid of the extra trait? It's super confusing